language/language.go

```diff
@@ -156,3 +156,171 @@ func DefaultLanguage() Language {
 
 	return ""
 }
+
+// LangID is a compact representation of a language
+// this package has orthographic knowledge of.
+//
+// The zero value represents a language not known by the package.
+type LangID uint16
+
+// NewLangID returns the compact index of the given language,
+// or false if it is not supported by this package.
+//
+// Derived languages not exactly supported are mapped to their primary part : for instance,
+// 'fr-be' is mapped to 'fr'
+func NewLangID(l Language) (LangID, bool) {
+	if i, ok := binarySearchLang(l, languagesInfos[:knownLangsCount]); ok {
+		return LangID(i), true
+	}
+	if i, ok := binarySearchLang(l, languagesInfos[knownLangsCount:]); ok {
+		return knownLangsCount + LangID(i), true
+	}
+	return 0, false
+}
+
+func binarySearchLang(l Language, records []languageInfo) (int, bool) {
```

Comment:
Could we use a battle-hardened binary search? I don't love re-writing this algorithm if we don't have to. I think [this](https://pkg.go.dev/slices#BinarySearchFunc) should be available in Go 1.19. If not, `golang.org/x/exp/slices` should be.

Reply:
`slices.BinarySearch` is only available in Go 1.21. Could we wait until we support it  ? Also, the implementation is really just a copy of what we had previously.

Reply:
Maybe https://pkg.go.dev/sort#Search could be used until Go 1.21 is the minimum version?